easypost/credit_card.py

```diff
@@ -0,0 +1,46 @@
+from typing import Optional
+
+from easypost.easypost_object import convert_to_easypost_object
+from easypost.error import Error
+from easypost.payment_method import PaymentMethod
+from easypost.requestor import (
+    RequestMethod,
+    Requestor,
+)
+from easypost.resource import (
+    CreateResource,
+    Resource,
+)
+
+
+class CreditCard(CreateResource, Resource):
+    @classmethod
+    def fund(cls, api_key: Optional[str] = None, **params) -> "CreditCard":
+        """Fund your EasyPost wallet by charging your primary or secondary card on file."""
+        payment_methods = PaymentMethod.all()
+
+        payment_method_map = {
+            "primary": "primary_payment_method",
+            "secondary": "secondary_payment_method",
+        }
+
+        primary_or_secondary = params.get("primary_or_secondary")
+        payment_method_to_use = payment_method_map.get(primary_or_secondary) if primary_or_secondary else None
+        card_id = payment_methods[payment_method_to_use]["id"] if payment_method_to_use else None
+
+        if payment_method_to_use is None or card_id is None or "card_" not in card_id:
+            raise Error(message="The chosen payment method is not a credit card. Please try again.")
+
+        requestor = Requestor(local_api_key=api_key)
+        url = f"{cls.class_url()}/{card_id}/charges"
+        wrapped_params = {"amount": params.get("amount")}
```

Comment:
I don't see the benefit of using `**params` in this function, can we just have two parameters in this function? 

Also In line 36, what if there is no `amount` in the `**params`? Should we check it before sending it to the API?

Reply:
Just following the pattern of the lib where we traditionally use `**params` for everything; however, I don't see the signature of this function changing much over time so we could use a more explicit keyword param approach.

Reply:
As for checking if it's empty etc, I don't feel we should here as there are additional validations needed at the API level anyway. Let's pass it through and make the API catch what's necessary all in one go.

Reply:
Got it. In Java, the param only needs `amount` so I wouldn't ask the user to input a Map in the parameter that only has one value.